extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts

```diff
@@ -558,116 +578,128 @@ class CornerstoneViewportService extends PubSubService
     // load any secondary displaySets
     const displaySetInstanceUIDs = this.viewportsDisplaySets.get(viewport.id);
 
-    const segDisplaySet = displaySetInstanceUIDs
+    // can be SEG or RTSTRUCT for now
+    const overlayDisplaySet = displaySetInstanceUIDs
       .map(displaySetService.getDisplaySetByUID)
-      .find(displaySet => displaySet && displaySet.Modality === 'SEG');
-
-    if (segDisplaySet) {
-      const { referencedVolumeId } = segDisplaySet;
-      const referencedVolume = cache.getVolume(referencedVolumeId);
-      const segmentationId = segDisplaySet.displaySetInstanceUID;
-
-      const toolGroup = toolGroupService.getToolGroupForViewport(viewport.id);
+      .find(displaySet => displaySet?.isOverlayDisplaySet);
```

Comment:
I like this handling a lot better - makes it much more consistent and extensible for other new overlay types.